go/trivial.go

```diff
@@ -3,33 +3,51 @@ package main
 import (
 	"fmt"
 	"os"
+	"strings"
 	"time"
 
-	"github.com/uber/jaeger-client-go"
+	jaeger "github.com/uber/jaeger-client-go"
 	"github.com/uber/jaeger-client-go/config"
 
-	"github.com/opentracing/opentracing-go"
+	opentracing "github.com/opentracing/opentracing-go"
 )
 
+func errExit(errstr string) {
+	fmt.Println(errstr)
+	os.Exit(1)
+}
+
 func main() {
+	var (
+		hostPort string
+	)
+
+	if len(os.Args) == 2 {
+		hostPort = os.Args[1]
+		if !strings.Contains(os.Args[1], ":") {
+			hostPort += ":6831"
+		}
+	}
+
 	// 1) Create a opentracing.Tracer that sends data to Zipkin
-	cfg := config.Configuration{
-		Sampler: &config.SamplerConfig{
-			Type:	"const",
-			Param:	1,
-		},
+	cfg := &config.Configuration{
 		Reporter: &config.ReporterConfig{
-			LogSpans:		true,
-			BufferFlushInterval:	1 * time.Second,
+			LogSpans:            true,
+			BufferFlushInterval: 1 * time.Second,
+			LocalAgentHostPort:  hostPort,
+		},
+		Sampler: &config.SamplerConfig{
+			Type:  "const",
+			Param: 1,
 		},
 	}
+
 	tracer, closer, err := cfg.New(
 		"your_service_name",
 		config.Logger(jaeger.StdLogger),
 	)
-	if (err != nil){
-		fmt.Println("Cannot create tracer: %v\n", err.Error())
-		os.Exit(1)
+	if err != nil {
+		errExit(fmt.Sprintf("Cannot create tracer: %+v\n", err))
```

Comment:
this is unnecessary complication of the example. Can be as simple as:
```go
panic("Cannot create tracer: " + err.Error())
```

Reply:
this is left over from coalescing an existing error check with the new one I added w/ fromenv and then later removed on request. I've refactored.